src/app/(auth)/login/page.tsx

```diff
@@ -42,26 +46,29 @@ export default function Login() {
       return;
     }
 
-    const { error } = await auth.signIn(email, password);
+    setIsLoggingIn(true);
+    const { error, data } = await auth.signIn(email, password);
 
     if (error) {
       setErrorMessage(error.message);
       // TODO: use error.status to check if it's an email-specific or password-specific error
       // then, raise the error in the TextInput component.
     } else {
-      profile?.loadProfile();
       setErrorMessage('');
-      setFinishLogin(true);
+      const loadedProfile = await profile.loadProfile(data.user?.id);
```

Comment:
just to clarify, would data.user ever be null if the signIn did not error? I'm assuming `data.user?` was added for typing purposes (i.e. since user is of type user | null)?

Reply:
i think its just to avoid type errors ye